python/plugins/processing/algs/qgis/EquidistanceBuffer.py

```diff
@@ -0,0 +1,169 @@
+        self.distance = None
+        self.segments = None
+        self.join_style = None
+        self.miter_limit = None
+        self.join_styles = [self.tr('Round'),
+                            self.tr('Miter'),
+                            self.tr('Bevel')]
+        self.end_cap_styles = [self.tr('Round'),
+                               self.tr('Flat'),
+                               self.tr('Square')]
+
+    def initParameters(self, config=None):
+        distanceParam = QgsProcessingParameterDistance(self.DISTANCE,
+                                                       self.tr('Distance'),
+                                                       defaultValue=10000.0)
+        distanceParam.setDefaultUnit(QgsUnitTypes.DistanceMeters)
+        self.addParameter(distanceParam)
+
+        self.addParameter(
+            QgsProcessingParameterNumber(
+                self.SEGMENTS,
+                self.tr('Segments'),
+                QgsProcessingParameterNumber.Integer,
+                minValue=1,
+                defaultValue=8))
+
+        self.addParameter(QgsProcessingParameterEnum(
+            self.END_CAP_STYLE,
+            self.tr('End cap style'),
+            options=self.end_cap_styles, defaultValue=0))
+
+        self.addParameter(QgsProcessingParameterEnum(
+            self.JOIN_STYLE,
+            self.tr('Join style'),
+            options=self.join_styles))
+
+        self.addParameter(
+            QgsProcessingParameterNumber(
+                self.MITER_LIMIT,
+                self.tr('Miter limit'),
+                QgsProcessingParameterNumber.Double,
+                minValue=1,
+                defaultValue=2))
+
+    def name(self):
+        return 'equidistancebuffer'
+
+    def displayName(self):
+        return self.tr('Equidistance buffer')
+
+    def outputName(self):
+        return self.tr('Buffer')
+
+    def inputLayerTypes(self):
+        return [QgsProcessing.TypeVector]
+
+    def outputType(self):
+        return QgsProcessing.TypeVectorPolygon
+
+    def outputWkbType(self, input_wkb_type):
+        return QgsWkbTypes.Polygon
+
+    def prepareAlgorithm(self, parameters, context, feedback):
+        self.distance = self.parameterAsDouble(
+            parameters, self.DISTANCE, context)
+        self.segments = self.parameterAsInt(parameters, self.SEGMENTS, context)
+        self.end_cap_style = self.parameterAsEnum(
+            parameters, self.END_CAP_STYLE, context) + 1
+        self.join_style = self.parameterAsEnum(
+            parameters, self.JOIN_STYLE, context) + 1
+        self.miter_limit = self.parameterAsDouble(
+            parameters, self.MITER_LIMIT, context)
+
+        source = self.parameterAsSource(parameters, 'INPUT', context)
+        self.source_crs = source.sourceCrs()
+        if not self.source_crs.isGeographic():
+            feedback.reportError(
+                self.tr(
+                    'Layer CRS must be a Geograhpic CRS for this algorithm'))
```

Comment:
```suggestion
                    'Layer CRS must be a Geographic CRS for this algorithm'))
```
and do we need the `for this algorithm` part?

Reply:
good point, it should be clear from the context that it concerns "this algorithm"

Reply:
fixed